### PR TITLE
DAOS-6890 control: improve consistency of dmg system command output

### PR DESCRIPTION
**Pull request by tanabarr**
* Report missing ranks either in action result table or separately and
  always print any missing hosts.
* Ensure dmg system commands return a non-zero exit code when either any
  delegated action fails e.g. rank stop or if any missing ranks or hosts
  were specified on the commandline.
* Rename some local variables to reflect naming change to I/O Engine.
* In dmg cmd execute methods, return early on api error and return
  errors from response otherwise.
* Move printer functions to pretty package and standardize use of io
  writer.

Signed-off-by: Tom Nabarro tom.nabarro@intel.com>

### CODE REVIEW

**Review by kjacque — DISMISSED** [review]
Hard to do a close inspection of the new changes since so much has been moved, but overall looks fine to me. One very minor comment if you have to change anything else.

File: src/control/system/membership.go [hunk 1]
```diff
@@ -444,9 +444,8 @@ func (m *Membership) MarkRankDead(rank Rank) error {
 	}
 
 	if member.State().isTransitionIllegal(MemberStateEvicted) {
-		m.log.Debugf("skipping illegal member state update for rank %d: %s->%s",
+		return errors.Errorf("llegal member state update for rank %d: %s->%s",
```
Comment (kjacque) [inline src/control/system/membership.go]: Minor typo in the error message, should probably fix that before landing

Comment (tanabarr) [reply src/control/system/membership.go]: fixing in https://github.com/daos-stack/daos/pull/4875

**Comment by mjmac** [comment]
Force-landing early after merging master and resolving a conflict. Changes since last successful run are only in the control plane, so no need to run full PR suite.

**Pull request merged**